packages/graphiql-toolkit/src/storage/history.ts

```diff
@@ -138,4 +144,31 @@ export class HistoryStore {
     }
     this.queries = [...this.history.items, ...this.favorite.items];
   }
+
+  deleteHistory = (
+    { query, variables, headers, operationName, favorite }: QueryStoreItem,
+    clearFavorites = false,
+  ) => {
+    function deleteFromStore(store: QueryStore) {
+      const found = store.items.find(
+        x =>
+          x.query === query &&
+          x.variables === variables &&
+          x.headers === headers &&
+          x.operationName === operationName,
+      );
+      if (found) {
+        store.delete(found);
+      }
+    }
+
+    if (favorite || clearFavorites) {
+      deleteFromStore(this.favorite);
+    }
+    if (!favorite || clearFavorites) {
+      deleteFromStore(this.history);
+    }
```

Comment:
something strange happens here, you will always call `deleteFromStore()` no matter `favorite` is truthy or falsy, so it can be simplified, or I am wrong?
```suggestion
      deleteFromStore(this.history);
```

Reply:
ah sorry, didn't saw that arguments - `this.favorite` and `this.history` are differents